napari/components/viewer_model.py

```diff
@@ -409,6 +409,43 @@ def add_pyramid(self, pyramid, *args, **kwargs):
         self.add_layer(layer)
         return layer
 
+    def add_volume(self, volume, *args, **kwargs):
+        """Adds an volume layer to the layers list.
+
+        Parameters
+        ----------
+        pyramid : list
+            3D array volume data. If the last dimensions of the images have length 3
+            or 4 they can be interpreted as RGB or RGBA if multichannel is
+            `True`.
+        metadata : dict, optional
+            Image metadata.
+        multichannel : bool, optional
+            Whether the image is multichannel RGB or RGBA if multichannel. If
+            not specified by user and the last dimension of the data has length
+            3 or 4 it will be set as `True`. If `False` the image is
+            interpreted as a luminance image.
+        colormap : str, vispy.Color.Colormap, 2-tuple, dict, optional
+            Colormap to use for luminance images. If a string must be the name
+            of a supported colormap from vispy or matplotlib. If a tuple the
+            first value must be a string to assign as a name to a colormap and
+            the second item must be a Colormap. If a dict the key must be a
+            string to assign as a name to a colormap and the value must be a
+            Colormap.
+
+        name : str, keyword-only
+            Name of the layer.
+
+        Returns
+        -------
+        layer : :class:`napari.layers.Volume`
+            The newly-created volume layer.
+        """
+        layer = layers.Volume(volume, *args, **kwargs)
+        self.dims = Dims(3)
```

Comment:
you should need this - instead for now I'd try
```python
layer = layers.Volume(volume, *args, **kwargs)
self.add_layer(layer)
self.dims.set_display(-2) = `True`
```
After you call the `set_display` the dims.indices should be
```python
(..., slice(None, None, None), slice(None, None, None), slice(None, None, None))
```
which will allow you to do the 3D slicing on the `Image` layer